tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeComponentImpl.kt

```diff
@@ -0,0 +1,126 @@
+package org.jetbrains.kotlin.gradle.plugin.experimental.internal
+
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.component.ComponentWithVariants
+import org.gradle.api.internal.component.SoftwareComponentInternal
+import org.gradle.api.internal.component.UsageContext
+import org.gradle.api.internal.file.FileOperations
+import org.gradle.api.internal.provider.LockableSetProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.internal.Describables
+import org.gradle.internal.DisplayName
+import org.gradle.language.ComponentDependencies
+import org.gradle.language.cpp.internal.NativeVariantIdentity
+import org.gradle.language.internal.DefaultBinaryCollection
+import org.gradle.language.internal.DefaultComponentDependencies
+import org.gradle.language.nativeplatform.internal.ComponentWithNames
+import org.gradle.language.nativeplatform.internal.DefaultNativeComponent
+import org.gradle.language.nativeplatform.internal.Names
+import org.gradle.language.nativeplatform.internal.PublicationAwareComponent
+import org.jetbrains.kotlin.gradle.plugin.experimental.KotlinNativeBinary
+import org.jetbrains.kotlin.gradle.plugin.experimental.KotlinNativeComponent
+import org.jetbrains.kotlin.gradle.plugin.experimental.sourcesets.KotlinNativeSourceSetImpl
+import org.jetbrains.kotlin.konan.target.HostManager
+import org.jetbrains.kotlin.konan.target.KonanTarget
+import javax.inject.Inject
+
+open class KotlinNativeComponentImpl @Inject constructor(
+        private val name: String,
+        override val sources: KotlinNativeSourceSetImpl,
+        private val objectFactory: ObjectFactory,
+        fileOperations: FileOperations
+) : DefaultNativeComponent(fileOperations), KotlinNativeComponent, ComponentWithNames, PublicationAwareComponent {
```

Comment:
According to the [Kotlin style guide](https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting) interfaces should be located in a new line

Reply:
Done